moveos/moveos-stdlib/moveos-stdlib/sources/object.move

```diff
@@ -2,159 +2,88 @@
-    public fun uid_as_inner(uid: &UID): &ID {
-        &uid.id
+    /// Borrow the object mutable value
+    public fun borrow_value_mut<T>(this: &mut Object<T>): &mut T{
+        &mut this.value
     }
 
-    /// Get the raw bytes of a `uid`'s inner `ID`
-    public fun uid_to_inner(uid: &UID): ID {
-        uid.id
-    }
+    /// ==== Object Store ====
 
-    /// Get the raw bytes of a `UID`
-    public fun uid_to_bytes(uid: &UID): vector<u8> {
-        bcs::to_bytes(&uid.id.bytes)
+    struct ObjectStore has drop{
+        table: any_table::Table<ObjectID>,
     }
 
-    /// Get the inner bytes of `id` as an address.
-    public fun uid_to_address(uid: &UID): address {
-        uid.id.bytes
+    public fun load_object_store(ctx: &TxContext): ObjectStore {
+        let table = any_table::load(tx_context::get_object_store_handle(ctx));
+        ObjectStore{table}
     }
 
-    // === any object ===
-
-    /// Create a new object. Returns the `UID` that must be stored in a Sui object.
-    /// This is the only way to create `UID`s.
-    public fun new(ctx: &mut TxContext): UID {
-        UID {
-            id: ID { bytes: tx_context::new_object(ctx) },
-        }
+    #[private_generic(T)]
+    public fun borrow<T: key>(this: &ObjectStore, object_id: ObjectID): &Object<T>{
+        any_table::borrow(&this.table, object_id)
     }
 
-    /// Delete the object and it's `UID`. This is the only way to eliminate a `UID`.
-    // This exists to inform Sui of object deletions. When an object
-    // gets unpacked, the programmer will have to do something with its
-    // `UID`. The implementation of this function emits a deleted
-    // system event so Sui knows to process the object deletion
-    public fun delete(id: UID) {
-        let UID { id: ID { bytes } } = id;
-        delete_impl(bytes)
+    #[private_generic(T)]
+    public fun borrow_mut<T: key>(this: &mut ObjectStore, object_id: ObjectID): &mut Object<T>{
+        any_table::borrow_mut(&mut this.table, object_id)
     }
-
-    /// Get the underlying `ID` of `obj`
-    public fun id<T: key>(obj: &T): ID {
-        borrow_uid(obj).id
+    
+    #[private_generic(T)]
+    /// Remove object from object store, only the owner can move the object
+    public fun remove<T: key>(this: &mut ObjectStore, object_id: ObjectID): Object<T>{
+        any_table::remove(&mut this.table, object_id)
     }
 
-    /// Borrow the underlying `ID` of `obj`
-    public fun borrow_id<T: key>(obj: &T): &ID {
-        &borrow_uid(obj).id
+    #[private_generic(T)]
+    public fun unpack<T>(obj: Object<T>): (ObjectID, T, address) {
+        let Object{id, value, owner} = obj;
+        (id, value, owner)
     }
 
-    /// Get the raw bytes for the underlying `ID` of `obj`
-    public fun id_bytes<T: key>(obj: &T): vector<u8> {
-        bcs::to_bytes(&borrow_uid(obj).id)
+    
+    #[private_generic(T)]
+    /// Add object to object store
+    public fun add<T: key>(this: &mut ObjectStore, obj: Object<T>) {
```

Comment:
why T need ability `key`? Does object stroage is independent of global storage ?

Reply:
I want to reuse the `key` ability to distinguish the object type and other types.

In Sui object, the object must have the `key` ability, and if have both `key` and `store`, the object can be public transfer.